connection.go

```diff
@@ -1163,3 +1193,71 @@ func (conn *Connection) NewStream() (*Stream, error) {
 		Conn: conn,
 	}, nil
 }
+
+// loadProtocolInfo sends info about client protocol,
+// receives info about server protocol in response
+// and store in in connection serverProtocolInfo.
+func (conn *Connection) loadProtocolInfo(w *bufio.Writer, r *bufio.Reader) error {
+	var resp Response
+	var err error
+
+	err = conn.writeProtocolInfoRequest(w, ClientProtocolVersion, ClientFeatures)
+	if err != nil {
+		return err
+	}
+
+	resp, err = conn.readProtocolInfoResponse(r)
+	if err != nil {
+		tarantoolError, ok := err.(Error)
+		if ok && tarantoolError.Code == ErrUnknownRequestType {
+			// IPROTO_ID requests are not supported by server.
+			conn.serverProtocolVersion = ProtocolVersionUnsupported
+			conn.serverFeatures = []Feature{}
+
+			return nil
+		}
+
+		return err
+	}
+
+	if len(resp.Data) == 0 {
+		return fmt.Errorf("Unexpected response on protocol info exchange: no data")
+	}
+
+	serverProtocolInfo, ok := resp.Data[0].(protocolInfo)
+	if !ok {
+		return fmt.Errorf("Unexpected response on protocol info exchange: wrong data")
+	}
+	conn.serverProtocolVersion = serverProtocolInfo.version
+	conn.serverFeatures = serverProtocolInfo.features
+
+	return nil
+}
+
+// ServerProtocolVersion returns protocol version supported by
+// connected Tarantool server.
+// Since 1.10.0
+func (conn *Connection) ServerProtocolVersion() ProtocolVersion {
+	return conn.serverProtocolVersion
+}
+
+// ClientProtocolVersion returns protocol version supported by
+// Go connection client.
+// Since 1.10.0
+func (conn *Connection) ClientProtocolVersion() ProtocolVersion {
+	return ClientProtocolVersion
+}
```

Comment:
Why not just use a constant ClientProtocolVersion? It looks like duplicate functionality.

Reply:
Let's discuss it here:

https://github.com/tarantool/go-tarantool/pull/226#discussion_r1022628684